analyzers/src/SonarAnalyzer.Common/Extensions/RegexExtensions.cs

```diff
@@ -42,23 +44,59 @@ public static Match SafeMatch(this Regex regex, string input)
     /// <summary>
     /// Matches the input to the regex. Returns <see langword="false" /> in case of an <see cref="RegexMatchTimeoutException" />.
     /// </summary>
-    public static bool SafeIsMatch(this Regex regex, string input)
+    public static bool SafeIsMatch(this Regex regex, string input) =>
+        regex.SafeIsMatch(input, false);
+
+    /// <summary>
+    /// Matches the input to the regex. Returns <paramref name="timeoutFallback"/> in case of an <see cref="RegexMatchTimeoutException" />.
+    /// </summary>
+    public static bool SafeIsMatch(this Regex regex, string input, bool timeoutFallback)
     {
         try
         {
             return regex.IsMatch(input);
         }
         catch (RegexMatchTimeoutException)
         {
-            return false;
+            return timeoutFallback;
         }
     }
 
-    public static bool SafeIsMatch(string input, string pattern, TimeSpan timeout, RegexOptions options = RegexOptions.None)
```

Comment:
I didn't notice that you worked on this as well and therefore fixed this slightly differently with what you did. The changes caused merge conflicts and I solved them by keeping my solution because I had a dedicated test method already and you method was in an extension class without being an extension method.

Reply:
Your implementation is better, I like it.